packages/protocol/contracts/governance/Election.sol

```diff
@@ -845,4 +885,40 @@ contract Election is
     }
     return res;
   }
+
+  struct SlashVotesInfo {
+    address[] groups;
+    uint256 remainingValue;
+  }
+
+  // Iterates in reverse order over all groups voted for by `account`,
+  // reducing the total amount of voting gold by `value` in total.
+  // For each group, if there `value` gold has left to be slashed, reduces
+  // the number of pending and active votes for that group by as much as
+  // possible without exceeding the total value that needs to be slashed.
+  function slashVotes(
+    address account,
+    uint256 value,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external nonReentrant returns (bool) {
```

Comment:
+1, should be `onlyRegisteredContract(LOCKED_GOLD_REGISTRY_ID)`

Normally, the way you would test this would be to register one of the ganache accounts as the `LockedGold` contract. The reason that doesn't work here is due to re-entrancy, this function is supposed to be called by `LockedGold`, and in turn calls *into* `LockedGold`. While there's nothing inherently wrong with this, some would consider it a dangerous pattern.

I see two options:
1. Don't rely on re-entrancy. Don't call `incrementNonVotingAccountBalance`, and let `LockedGold` do that based on the return value of `slashVotes` (i.e. have it return a uint256).
2. Keep things re-entrant, and define `_slashVotes` which is internal and does not have the modifier. Then, expose it in `ElectionTest` the same way we do for `onlyVm` functions.


@mrsmkl @lucasege which do you prefer?

Reply:
I think I'd prefer 1, but maybe we also change the name from `slashVotes` to just `decrementVotes` or something that indicates we are only decrementing and not actually sending rewards yet (since increment and sending are done in lockedgold).

Reply:
I like `decrementVotes`